File: lead-generator/shared/models/campaign.py
from datetime import datetime
from typing import Any
from uuid import UUID, uuid4

from pydantic import BaseModel, Field


class CampaignStatus(str):
    DRAFT = "draft"
    ACTIVE = "active"
    PAUSED = "paused"
    FINISHED = "finished"


class Campaign(BaseModel):
    id: UUID = Field(default_factory=uuid4)
    name: str
    slug: str
    status: str = "draft"
    objective: str | None = None
    source_config: dict[str, Any] = Field(default_factory=dict)
    is_active: bool = True
    # Campos de oferta — usados pelo orquestrador em modo varredura
    offer_description: str | None = None
    ideal_customer_profile: str | None = None
    ticket: str | None = None
    focus_segments: list[str] = Field(default_factory=list)
    created_at: datetime = Field(default_factory=datetime.utcnow)

File: api/src/MotorAudiencia.Domain/Entities/Campaign.cs
// Domain/Entities/Campaign.cs
namespace MotorAudiencia.Domain.Entities;

public sealed class Campaign
{
    public Guid Id { get; private set; } = Guid.NewGuid();
    public string Name { get; private set; } = string.Empty;
    public string Slug { get; private set; } = string.Empty;
    public string Status { get; private set; } = "draft";
    public string? OfferDescription { get; private set; }
    public string? IdealCustomerProfile { get; private set; }
    public string? Ticket { get; private set; }
    private List<string> _keywordsAlvo = [];
    public IReadOnlyList<string> KeywordsAlvo => _keywordsAlvo.AsReadOnly();
    public bool IsActive { get; private set; } = true;
    public DateTime CreatedAt { get; private set; } = DateTime.UtcNow;

    private Campaign() { }

    public static Campaign Create(string name, string slug)
    {
        ArgumentException.ThrowIfNullOrWhiteSpace(name);
        ArgumentException.ThrowIfNullOrWhiteSpace(slug);
        return new Campaign { Name = name, Slug = slug };
    }
}
